dataprofiler/labelers/column_name_model.py

```diff
@@ -0,0 +1,232 @@
+from .base_model import AutoSubRegistrationMeta, BaseModel
+
+logger = dp_logging.get_child_logger(__name__)
+
+
+class ColumnNameModel(BaseModel, metaclass=AutoSubRegistrationMeta):
+    """Class for column name data labeling model."""
+
+    def __init__(self, label_mapping=None, parameters=None):
+        r"""
+        :param label_mapping: maps labels to their encoded integers
+        :type label_mapping: dict
+        :param parameters: Contains all the appropriate parameters for the model.
+            Possible parameters are:
+                max_length, max_num_chars, dim_embed
+        :type parameters: dict
+        :return: None
+        """
+        # parameter initialization
+        if not parameters:
+            parameters = {}
+        parameters.setdefault('false_positive_dict', None)
+        parameters.setdefault('true_positive_dict', None)
+
+        # initialize class
+        self.set_label_mapping(label_mapping)
+        self._validate_parameters(parameters)
+        self._parameters = parameters
+
+    def _validate_parameters(self, parameters):
+        r"""
+        Validate the parameters sent in.
+
+        Raise error if invalid parameters are present.
+
+        :param parameters: parameter dict containing the following parameters:
+            true_positive_dict
+            false_positive_dict
+        :type parameters: dict
+        :return: None
+        """
+        errors = []
+
+        list_of_accepted_parameters = [
+            "true_positive_dict",
+            "false_positive_dict",
+        ]
+
+        for param in parameters:
+            value = parameters[param]
+            if param == "false_positive_dict" and ( 
+                not isinstance(value, list)
+                or 'attribute' not in value[0].keys()
+            ):
+                errors.append(
+                    "`{}` must be a list of dictionaries with at least 'attribute' as the key".format(param)
+                )
+            elif param == "true_positive_dict" and (
+                not isinstance(value, list)
+                or not isinstance(value[0], dict)
+                or not isinstance(value[1]['attribute'], str)
+                or not isinstance(value[1]['label'], str)
+            ):
+                errors.append(
+                    """`{}` must be a list of dictionaries each with the following
+                    two keys: 'attribute' and 'label'""".format(param)
+                )
+            elif param not in list_of_accepted_parameters:
+                errors.append("`{}` is not an accepted parameter.".format(param))
+        if errors:
+            raise ValueError("\n".join(errors))
+
+    def _compare_negative(self, list_of_column_names, check_values_dict, negative_threshold):
+        """Filter out column name examples that are false positives"""
+        scores = self._model(
+                list_of_column_names,
+                check_values_dict,
+                self._make_lower_case(),
```

Comment:
is this something we are going to parameterize? why not just do `str.lower`?

Reply:
Good question -- thought the same myself. The parameter for the `process.cdist` function must be `callable` so that is a big requirement. `str.lower()` isn't callable since that returns a `str` object type. 

Documentation to the `process.cdist` function [here](https://maxbachmann.github.io/RapidFuzz/Usage/process.html#cdist)